tests/functional/base_test.go

```diff
@@ -94,37 +96,78 @@ func CreateOVNDBCluster(namespace string, OVNDBClusterName string, spec map[stri
 	return th.CreateUnstructured(raw)
 }
 
+func UpdateOVNDBCluster(cluster *ovnv1.OVNDBCluster) {
+	k8sClient.Update(ctx, cluster)
+}
+
 func OVNDBClusterConditionGetter(name types.NamespacedName) condition.Conditions {
 	instance := GetOVNDBCluster(name)
 	return instance.Status.Conditions
 }
 
+func ScaleDBCluster(name types.NamespacedName, replicas int32) {
+	Eventually(func(g Gomega) {
+		c := GetOVNDBCluster(name)
+		*c.Spec.Replicas = replicas
+		g.Expect(k8sClient.Update(ctx, c)).Should(Succeed())
+	}).Should(Succeed())
+}
+
 // CreateOVNDBClusters Creates NB and SB OVNDBClusters
-func CreateOVNDBClusters(namespace string, nad string) []types.NamespacedName {
+func CreateOVNDBClusters(namespace string, nad map[string][]string, replicas int32) []types.NamespacedName {
 	dbs := []types.NamespacedName{}
 	for _, db := range []string{v1beta1.NBDBType, v1beta1.SBDBType} {
 		name := fmt.Sprintf("ovn-%s", uuid.New().String())
 		spec := GetDefaultOVNDBClusterSpec()
+		stringNad := ""
+		// OVNDBCluster doesn't allow multiple NADs, hence map len
+		// must be <= 1
+		Expect(len(nad)).Should(BeNumerically("<=", 1))
+		for k, _ := range nad {
+			if strings.Contains(k, "/") {
+				// k = namespace/nad_name, split[1] will return nad_name (e.g. internalapi)
+				stringNad = strings.Split(k, "/")[1]
+			}
+		}
+		if len(nad) != 0 {
+			// nad format needs to be map[string][]string{namespace + "/" + nad_name: ...} or empty
+			Expect(stringNad).ToNot(Equal(""))
+		}
 		spec["dbType"] = db
-		spec["networkAttachment"] = nad
+		spec["networkAttachment"] = stringNad
+		spec["replicas"] = replicas
 
 		instance := CreateOVNDBCluster(namespace, name, spec)
 
 		instance_name := types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()}
 
-		dbaddr := "tcp:10.1.1.1:6641"
+		dbName := "nb"
 		if db == v1beta1.SBDBType {
-			dbaddr = "tcp:10.1.1.1:6642"
+			dbName = "sb"
+		}
+		statefulSetName := types.NamespacedName{
+			Namespace: instance.GetNamespace(),
+			Name:      "ovsdbserver-" + dbName,
 		}
-		// the Status field needs to be written via a separate client
+		th.SimulateStatefulSetReplicaReadyWithPods(
+			statefulSetName,
+			nad,
+		)
+		// Ensure that PODs are ready and DBCluster have been reconciled
+		// with all information (Status.DBAddress and internalDBAddress
+		// are set at the end of the reconcileService)
 		Eventually(func(g Gomega) {
 			ovndbcluster := GetOVNDBCluster(instance_name)
-			ovndbcluster.Status.InternalDBAddress = dbaddr
-			if nad != "" {
-				ovndbcluster.Status.DBAddress = dbaddr
+			endpoint := ""
+			// If cluster NAD is empty, ExtenralEndpoint will never be
+			// set, instead look at internalEndpoint
```

Comment:
I don't think we should check "instead". internal can always be checked; external when nad is set.